MSAL/src/native_auth/controllers/sign_in/MSALNativeAuthSignInController.swift

```diff
@@ -358,13 +319,222 @@ final class MSALNativeAuthSignInController: MSALNativeAuthTokenController, MSALN
+            includeChallengeType: false,
+            context: context) else {
+            MSALLogger.log(level: .error, context: context, format: "Submit code: unable to create token request")
+
+            return processSubmitCodeFailure(
+                errorType: .generalError(nil),
+                telemetryInfo: telemetryInfo,
+                scopes: scopes,
+                continuationToken: continuationToken,
+                context: context
+            )
+        }
+        let config = factory.makeMSIDConfiguration(scopes: scopes)
+        let response = await performAndValidateTokenRequest(request, config: config, context: context)
+        switch response {
+        case .success(let tokenResponse):
+            return await withCheckedContinuation { continuation in
+                handleMSIDTokenResponse(
+                    tokenResponse: tokenResponse,
+                    context: context,
+                    telemetryInfo: telemetryInfo,
+                    config: config,
+                    onSuccess: { accountResult in
+                        continuation.resume(
+                            returning: .init(
+                                .completed(accountResult),
+                                correlationId: context.correlationId(),
+                                telemetryUpdate: { [weak self] result in
+                            self?.stopTelemetryEvent(telemetryInfo.event, context: context, delegateDispatcherResult: result)
+                        }))
+                    },
+                    onError: { [weak self] error in
+                        MSALLogger.logPII(
+                            level: .error,
+                            context: context,
+                            format: "Submit code, token request failed with error \(MSALLogMask.maskPII(error.errorDescription))"
+                        )
+                        guard let self = self else { return }
+                        continuation.resume(returning: self.processSubmitCodeFailure(
+                            errorType: .generalError(nil),
+                            telemetryInfo: telemetryInfo,
+                            scopes: scopes,
+                            continuationToken: continuationToken,
+                            context: context
+                        ))
+                    }
+                )
+            }
+        case .error(let errorType):
+            return processSubmitCodeFailure(
+                errorType: errorType,
+                telemetryInfo: telemetryInfo,
+                scopes: scopes,
+                continuationToken: continuationToken,
+                context: context
+            )
+        case .strongAuthRequired(_):
+            let error = VerifyCodeError(type: .generalError, correlationId: context.correlationId())
+            MSALLogger.log(level: .error, context: context, format: "Submit code: received unexpected MFA required API result")
```

Comment:
I don't think is useful here because we need to inform the user that an unexpected server response has been received. The error description that we're printing at line 851 is an internal error description, not the one from the server